fifo.go

```diff
@@ -147,7 +152,7 @@ func OpenFifo(ctx context.Context, fn string, flag int, perm os.FileMode) (io.Re
 // Read from a fifo to a byte array.
 func (f *fifo) Read(b []byte) (int, error) {
 	if f.flag&syscall.O_WRONLY > 0 {
-		return 0, errors.New("reading from write-only fifo")
+		return 0, ReadWrOnlyFifoError
```

Comment:
should probably add unit tests for these over in https://github.com/containerd/fifo/blob/master/raw_test.go